src/server.js

```diff
@@ -388,6 +388,34 @@ function initializeClient(socket, client, token, lastMessage) {
 
 	socket.on("sessions:get", sendSessionList);
 
+	socket.on("setting:set", (newSetting) => {
+		// Older user configs will not have the clientSettings property.
+		if (!client.config.hasOwnProperty("clientSettings")) {
+			client.config.clientSettings = {};
+		}
+		// We do not need to do write operations and emit events if nothing changed.
+		if (client.config.clientSettings[newSetting.name] !== newSetting.value) {
+			const newclientSettings = client.config.clientSettings;
+			newclientSettings[newSetting.name] = newSetting.value;
+			client.manager.updateUser(client.name, {
+				clientSettings: newclientSettings,
+			});
+			// Pass the setting to all clients.
+			client.emit("setting:new", {
+				name: newSetting.name,
+				value: newSetting.value,
+			});
+		}
+	});
+
+	socket.on("settings:get", () => {
+		if (!client.config.hasOwnProperty("clientSettings")) {
+			socket.emit("setting:all", {});
```

Comment:
You forgot a return.

Reply:
Actually forgot the else, but return works as well. Fixing that now.